Simplified/AxisService/NYPLAxisItemDownloader.swift

```diff
@@ -92,6 +94,18 @@ class NYPLAxisItemDownloader: NYPLAxisItemDownloading {
           self.leaveGroupAndStopDownload()
         }
       case .failure(let error):
+        if (error as NSError).code == NSURLErrorCancelled {
+          self.dispatchGroup.leave()
+          self.delegate = nil
+          self.downloadProgressHandler.progressListener = nil
+          return
+        }
+        
+        if !self.shouldContinue {
+          self.dispatchGroup.leave()
```

Comment:
shouldContinue will only be false after `leaveGroupAndStopDownload()` gets executed. When it does, it nils the `delegate` and `progressListener`

Reply:
the behavior you described is a little hard to see IMO -- if I understand correctly we are relying on the fact that `leaveGroupAndStopDownload` sets `isTerminated` to true which then causes `shouldContinue` to be false, and so at that point delegate and progressListener are already false because of what happens in `leaveGroupAndStopDownload`.  If so then I get it -- however, as things are now this is not a contract, and so if in the future we change the behavior of `leaveGroupAndStopDownload` this implicit behavior might cause a bug. 

Another thing that could happen is that if `shouldContinue` is not mapped one-to-one with `isTerminated`, then `shouldContinue` will no longer be false if `isTerminated` is true.

We could make it more explicit by specifying that the delegate and progressListener properties are only alive while a download is in progress, and here we could just check for isTerminated instead of shouldContinue.